src/openforms/variables/utils.py

```diff
@@ -0,0 +1,27 @@
+from datetime import date, datetime, time
+from typing import Any
+
+
+def check_date(value: str) -> str:
+    try:
+        date.fromisoformat(value)
+    except ValueError:
+        datetime.fromisoformat(value)
+    return value
+
+
+def check_time(value: str) -> str:
+    try:
+        time.fromisoformat(value)
+    except ValueError:
+        datetime.fromisoformat(value)
+    return value
+
+
+def check_initial_value(initial_value: Any, data_type: str) -> Any:
```

Comment:
`Any` is a bit too to broad - you are limited to `JSONValue` since it has to be serialized as JSON.